lbz/misc.py

```diff
@@ -132,3 +133,25 @@ def deep_update(dict_to_update: dict, update_data: dict) -> None:
 
 def is_in_debug_mode() -> bool:
     return environ.get("LBZ_DEBUG_MODE", "").lower() == "true"
+
+
+def deprecated(func: Callable) -> Any:
+    """
+    This is a decorator which can be used to mark functions as deprecated.
+    It will result in a warning being emitted when the function is used.
+
+    Copied from https://stackoverflow.com/a/30253848
```

Comment:
Personally, I like more what e.g. [deprecation](https://pypi.org/project/deprecation/) or [Deprecated](https://pypi.org/project/Deprecated/) gives. I mean that we can specify why the method/function is deprecated and point to the version when it will be removed.

I don't think we need to use a library for that but it would be good to make this code snippet better if we really want to start using it (tests welcome) 😉